packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -1068,113 +979,102 @@ export const useArbTokenBridge = (
 
   async function triggerOutboxToken({
     id,
-    l1Signer
+    l1Signer,
+    txLifecycle
   }: {
     id: string
     l1Signer: Signer
+    txLifecycle: TokenTriggerOutboxTransactionLifecycle
   }) {
     const event = pendingWithdrawalsMap[id]
 
     if (!pendingWithdrawalsMap[id]) {
       throw new Error('Outbox message not found')
     }
 
-    const { tokenAddress, value } = event
-
-    const messageWriter = L2ToL1Message.fromEvent(
-      l1Signer,
-      event,
-      await getOutboxAddress(event)
-    )
-
-    const res = await messageWriter.execute(l2.provider)
-
-    const { symbol, decimals } = await getL1TokenData(tokenAddress as string)
-
-    addTransaction({
-      status: 'pending',
-      type: 'outbox',
-      value: utils.formatUnits(value, decimals),
-      assetName: symbol,
-      assetType: AssetType.ERC20,
-      sender: walletAddress,
-      txID: res.hash,
-      l1NetworkID,
-      l2ToL1MsgData: { uniqueId: getUniqueIdOrHashFromEvent(event) }
-    })
-
     try {
-      const rec = await res.wait()
+      const messageWriter = L2ToL1Message.fromEvent(
+        l1Signer,
+        event,
+        await getOutboxAddress(event)
+      )
+      const tx = await messageWriter.execute(l2.provider)
+
+      const tokenData = await getL1TokenData(event.tokenAddress as string)
+      if (txLifecycle?.onTxSubmit) {
+        txLifecycle.onTxSubmit(tx, event, tokenData)
+      }
+      const receipt = await tx.wait()
 
-      if (rec.status === 1) {
-        setTransactionSuccess(rec.transactionHash)
+      if (receipt.status === 1) {
```

Comment:
Can we use an enum here, or add a comment maybe. What does that 1 stand for?

Reply:
that's a good point
i renamed `rec` to `receipt` here for readability reason
let me look into what `1` is 😂